src/problem3.rb

```diff
@@ -0,0 +1,33 @@
+class Problem3
+  def validateAndGetDuplicateValues(values)
+    @response = []
+    if(!values.nil? && (values.instance_of?(Array)))
+      if(values.all? {|i| i.is_a?(Integer)})
```

Comment:
The problem states `Try to find a solution by executing just one loop.`

What do you think the implementation of `all?` does? Hint: when you click on the method header in the ruby docs it will show you the c implementation.

Reply:
Well the solution is actually done in only one loop, that's just a pre-validation, probably not needed if we actually always provide integers. getDuplicateValues is the function that does all the work.